src/MultiFormats/Cryptography/Keccak.cs

```diff
@@ -0,0 +1,166 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+/*
+ * The package was named SHA3 it is really Keccak
+ * https://medium.com/@ConsenSys/are-you-really-using-sha-3-or-old-code-c5df31ad2b0
+ * See 
+ * 
+ * The SHA3 package doesn't create .Net Standard package.
+ * This is a copy of https://bitbucket.org/jdluzen/sha3/raw/d1fd55dc225d18a7fb61515b62d3c8f164d2e788/SHA3/SHA3.cs
+ */
+
+using System;
+
+namespace MultiFormats.Cryptography
+{
+    internal abstract class Keccak : System.Security.Cryptography.HashAlgorithm
+    {
+        #region Implementation
+
+        public const int KeccakB = 1600;
+        public const int KeccakNumberOfRounds = 24;
+        public const int KeccakLaneSizeInBits = 8 * 8;
+
+        public readonly ulong[] RoundConstants;
+
+        protected ulong[] state;
+        protected byte[] Buffer;
+        protected int BuffLength;
+
+        protected int keccakR;
+
+        public int KeccakR { get => keccakR; protected set => keccakR = value; }
+
+        public int SizeInBytes => KeccakR / 8;
+
+        public int HashByteLength => HashSizeValue / 8;
+
+        public override bool CanReuseTransform => true;
+
+        protected Keccak(int hashBitLength)
+        {
+            if (hashBitLength != 224 && hashBitLength != 256 && hashBitLength != 384 && hashBitLength != 512)
+            {
+                throw new ArgumentException("hashBitLength must be 224, 256, 384, or 512", nameof(hashBitLength));
+            }
+            
+            Initialize();
+            HashSizeValue = hashBitLength;
+            switch (hashBitLength)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=5051852)